chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/rules/developer/DeveloperStatusChangedValidation.java

```diff
@@ -0,0 +1,59 @@
+package gov.healthit.chpl.manager.rules.developer;
+
+import org.apache.commons.lang.StringUtils;
+
+import gov.healthit.chpl.dao.DeveloperDAO;
+import gov.healthit.chpl.dto.DeveloperDTO;
+import gov.healthit.chpl.dto.DeveloperStatusEventDTO;
+import gov.healthit.chpl.entity.developer.DeveloperStatusType;
+import gov.healthit.chpl.manager.impl.DeveloperManagerImpl;
+import gov.healthit.chpl.manager.rules.ValidationRule;
+import gov.healthit.chpl.permissions.ResourcePermissions;
+import gov.healthit.chpl.util.ErrorMessageUtil;
+import gov.healthit.chpl.util.ValidationUtils;
+
+public class DeveloperStatusChangedValidation extends ValidationRule<DeveloperValidationContext> {
+    private ResourcePermissions resourcePermissions;
+
+    public DeveloperStatusChangedValidation(final ResourcePermissions resourcePermissions) {
+        this.resourcePermissions = resourcePermissions;
+    }
+
+    public DeveloperStatusChangedValidation(final DeveloperManagerImpl developerManagerImpl,
+            final ResourcePermissions resourcePermissions, final DeveloperDAO developerDao) {
+        this.resourcePermissions = resourcePermissions;
+    }
+
+    /**
+     * determine if the status has been changed in most cases only allowed by
+     * ROLE_ADMIN but ROLE_ACB can change it to UnderCertificationBanByOnc
+     */
+    @Override
+    public boolean isValid(DeveloperValidationContext context) {
+        ErrorMessageUtil msgUtil = context.getErrorMessageUtil();
+        DeveloperDTO updatedDev = context.getDeveloperDTO();
+        DeveloperDTO beforeDev = context.getBeforeDev();
+        DeveloperStatusEventDTO newDevStatus = updatedDev.getStatus();
+        DeveloperStatusEventDTO currDevStatus = beforeDev.getStatus();
+
+        boolean currentStatusChanged = !currDevStatus.getStatus().getStatusName()
+                .equals(newDevStatus.getStatus().getStatusName());
+        if (currentStatusChanged
+                && !newDevStatus.getStatus().getStatusName()
+                        .equals(DeveloperStatusType.UnderCertificationBanByOnc.toString())
+                && !resourcePermissions.isUserRoleAdmin() && !resourcePermissions.isUserRoleOnc()) {
+            String msg = msgUtil.getMessage("developer.statusChangeNotAllowedWithoutAdmin");
+            getMessages().add(msg);
+            return false;
+        }
+        // Check to see that the Developer's website is valid.
+        if (!StringUtils.isEmpty(updatedDev.getWebsite())) {
```

Comment:
What does the website have to do with the status change? Should be extracted

Reply:
In `public ValidationRule<DeveloperValidationContext> getRule(String name) {` there's already a case for "Website required"

Reply:
Yeah I agree - I don't know why it was there.  It used to be an an else condition so intended to be a part of the same routine by original writer so I left together (but removed else as redundant in return/exception type situation). Assuming there is no valid reason for it, If that exact validation is already run prior, then I won't call again. If it is not run, I'll call it externally. Either way, will remove from here.

Reply:
There's 2 different website validations. Well-formed and required. Well-formed is called by update where as only system validations use required. This extra code represents the well-formed pattern and validates of the same version (updated) of the developer and therefore can be removed altogether/doesn't need to be repeated/externalized. I will remove it.